Add customizable meeting reminders

antopalidi opened the pull request:
#### :tophat: What? Why?
This PR enhances the automatic meeting reminders in Decidim by making them more customizable. Previously, registered participants received a reminder exactly 48 hours before the meeting. Now, administrators and meeting organizers can:

- Choose whether a reminder should be sent at all.
- Customize the reminder timing (set how many hours before the meeting the notification should be sent).
- Edit the reminder content using a text editor, similar to invitation emails.

#### :pushpin: Related Issues
- Fixes #80 

#### Testing
1. **Create a meeting (admin or public)**  
   - Create a meeting and enable reminders.  
   - Set the reminder time (e.g., 72 hours before the start).  
   - Add a custom message.  
   - Ensure that there are registered participants (followers) for the meeting.  

2. **Verify system notification**  
   - Check that a notification appears in the user’s notification panel.  

3. **Verify email delivery**  
   - Confirm that participants receive an email reminder.  
   - Ensure the email includes the correct time and custom message.  

### :camera: Screenshots
<img width="800" alt="Screenshot 2025-02-06 at 11 02 00" src="https://github.com/user-attachments/assets/75f48f0f-0e78-4683-9992-853f42e58507" />

:hearts: Thank you!

antopalidi committed:
add form to admin

antopalidi committed:
add migration, change event

antopalidi committed:
add new fields to public

antopalidi committed:
add admin form test

antopalidi committed:
add meetings form tests

antopalidi committed:
change commands

antopalidi committed:
fix commands, add tests

antopalidi committed:
fix commands, add tests

antopalidi committed:
fix rubocop

antopalidi committed:
Merge branch 'develop' into feature/meeting-reminder-customization

antopalidi committed:
fix event

antopalidi committed:
change label in a form

microstudi commented on decidim-meetings/config/locales/en.yml:
```diff
@@ -360,9 +360,14 @@ en:
             online_meeting_url_help: 'Link: allow participants to connect directly to your meeting'
             registration_url_help: 'Link: allow participants to go on the external service you are using for registrations'
+            reminder_enabled: Reminder enabled
+            reminder_help_text: Participants will receive an email the number of hours specified in advance of themeeting. Leave it empty or zero if no reminder is necessary.
```
```suggestion
            reminder_help_text: Participants will receive an email the number of hours specified in advance of themeeting. If left empty or zero, the reminder will be sent at the default configured value of %{default_value} hours.
```

microstudi replied:
@antopalidi the client has suggested that this should be the expected behavior. So let's change it to accomplish this in the rest of the code. Add a test case as well.

antopalidi committed:
change validations with send_reminders_before_hours is zero or nil

microstudi reviewed (changes requested):
Note: there is a bug when publishing the meeting and you don't set the "custom reminders feature"!

antopalidi committed:
fix error

antopalidi committed:
fix error

antopalidi committed:
add test

antopalidi commented:
> Note: there is a bug when publishing the meeting and you don't set the "custom reminders feature"!

@microstudi 
I fixed this problem and added some system tests for this.

antopalidi committed:
update commands

antopalidi committed:
fix commands
